Mathlib/Tactic/NormNum/LegendreSymbol.lean

```diff
@@ -0,0 +1,459 @@
+            ⟨er, q(jacobiSymNat.even_odd₁ $ea $eb $ec $er $ha $hb $hc $p)⟩
+          | 3 =>
+            have er' := mkRawIntLit (-er.intLit!)
+            have hb : Q(Nat.mod $eb 8 = 3) := (q(Eq.refl 3) : Expr)
+            show (_ : Q(ℤ)) × Q(jacobiSymNat $ea $eb = -$er) from
+              ⟨er', q(jacobiSymNat.even_odd₃ $ea $eb $ec $er $ha $hb $hc $p)⟩
+          | 5 =>
+            have er' := mkRawIntLit (-er.intLit!)
+            have hb : Q(Nat.mod $eb 8 = 5) := (q(Eq.refl 5) : Expr)
+            show (_ : Q(ℤ)) × Q(jacobiSymNat $ea $eb = -$er) from
+              ⟨er', q(jacobiSymNat.even_odd₅ $ea $eb $ec $er $ha $hb $hc $p)⟩
+          | _ =>
+            have hb : Q(Nat.mod $eb 8 = 7) := (q(Eq.refl 7) : Expr)
+            ⟨er, q(jacobiSymNat.even_odd₇ $ea $eb $ec $er $ha $hb $hc $p)⟩
+      | _ =>
+        have eab : Q(ℕ) := mkRawNatLit (b % a)
+        have hab : Q(Nat.mod $eb $ea = $eab) := (q(Eq.refl $eab) : Expr)
+        have ⟨er, p⟩ := proveJacobiSymOdd eab ea
+        match a % 4 with
+        | 1 =>
+          have ha : Q(Nat.mod $ea 4 = 1) := (q(Eq.refl 1) : Expr)
+          have hb : Q(Nat.mod $eb 2 = 1) := (q(Eq.refl 1) : Expr)
+          ⟨er, q(jacobiSymNat.qr₁_mod $ea $eb $eab $er $ha $hb $hab $p)⟩
+        | _ =>
+          match b % 4 with
+          | 1 =>
+            have ha : Q(Nat.mod $ea 2 = 1) := (q(Eq.refl 1) : Expr)
+            have hb : Q(Nat.mod $eb 4 = 1) := (q(Eq.refl 1) : Expr)
+            ⟨er, q(jacobiSymNat.qr₁'_mod $ea $eb $eab $er $ha $hb $hab $p)⟩
+          | _ =>
+            have er' := mkRawIntLit (-er.intLit!)
+            have ha : Q(Nat.mod $ea 4 = 3) := (q(Eq.refl 3) : Expr)
+            have hb : Q(Nat.mod $eb 4 = 3) := (q(Eq.refl 3) : Expr)
+            show (_ : Q(ℤ)) × Q(jacobiSymNat $ea $eb = -$er) from
+              ⟨er', q(jacobiSymNat.qr₃_mod $ea $eb $eab $er $ha $hb $hab $p)⟩
+#align norm_num.prove_jacobi_sym_odd Mathlib.Meta.NormNum.proveJacobiSymOdd
+
+/-- This evaluates `r := jacobiSymNat a b` and produces a proof term for the equality
+by removing powers of `2` from `b` and then calling `proveJacobiSymOdd`. -/
+partial def proveJacobiSymNat (ea eb : Q(ℕ)) : (er : Q(ℤ)) × Q(jacobiSymNat $ea $eb = $er) :=
+  match eb.natLit! with
+  | 0 =>
+    show (er : Q(ℤ)) × Q(jacobiSymNat $ea 0 = $er) from
+      ⟨mkRawIntLit 1, q(jacobiSymNat.zero_right $ea)⟩
+  | 1 =>
+    show (er : Q(ℤ)) × Q(jacobiSymNat $ea 1 = $er) from
+      ⟨mkRawIntLit 1, q(jacobiSymNat.one_right $ea)⟩
+  | b =>
+    match b % 2 with
+    | 0 =>
+      match ea.natLit! with
+      | 0 =>
+        have hb : Q(Nat.beq ($eb / 2) 0 = false) := (q(Eq.refl false) : Expr)
+        show (er : Q(ℤ)) × Q(jacobiSymNat 0 $eb = $er) from
+          ⟨mkRawIntLit 0, q(jacobiSymNat.zero_left $eb $hb)⟩
+      | 1 =>
+        show (er : Q(ℤ)) × Q(jacobiSymNat 1 $eb = $er) from
+          ⟨mkRawIntLit 1, q(jacobiSymNat.one_left $eb)⟩
+      | a =>
+        match a % 0 with
```

Comment:
This looks incorrect. I guess it should be `a % 2` instead of `a % 0`.

Reply:
Oh, that's wrong, thank you.
I'll create the PR after porting `Traversable` deriving handler.

Reply:
#5719